lib/protein/service.rb

```diff
@@ -64,7 +64,15 @@ def process
     @response = response_class.new if response?
     @errors = []
 
-    call
+    begin
+      call
+    rescue Exception => ex
+      Protein.config.error_logger&.call "#{self.class} failed with exception: #{ex.inspect}"
+      if response?
+        add_error(ex.message)
```

Comment:
Perhaps the error argument could be moar formal, sth along `:uncaught_exception, ex.message`, WDYT?
/cc @bartekbsh